src/query/predicate.cc

```diff
@@ -340,27 +342,34 @@ EvaluationResult EvaluateTextPredicate(const Predicate* predicate,
 // requirements.
 EvaluationResult ComposedPredicate::Evaluate(Evaluator& evaluator) const {
   if (GetType() == PredicateType::kComposedAnd) {
+    // Short-circuit on first false
+    uint32_t childPositionsCount = 0;
+    uint64_t query_field_mask = ~0ULL;
+    std::vector<std::unique_ptr<indexes::text::TextIterator>> iterators;
+    // Determine if children need to return positions for proximity checks
     bool require_positions = slop_.has_value() || inorder_;
-    EvaluationResult lhs = EvaluateTextPredicate(lhs_predicate_.get(),
-                                                 evaluator, require_positions);
-    // Short-circuit for AND
-    if (!lhs.matches) return EvaluationResult(false);
-    EvaluationResult rhs = EvaluateTextPredicate(rhs_predicate_.get(),
-                                                 evaluator, require_positions);
-    // Short-circuit for AND
-    if (!rhs.matches) return EvaluationResult(false);
-    // Proximity check: Only if slop/inorder set and both sides have iterators
-    if (require_positions && lhs.filter_iterator && rhs.filter_iterator) {
+    for (const auto& child : children_) {
+      EvaluationResult result =
+          EvaluateTextPredicate(child.get(), evaluator, require_positions);
+      if (result.filter_iterator && require_positions) {
```

Comment:
```suggestion
      if (result.filter_iterator) {
```
Checking for `require_positions` here is redundant